src/main/java/ar/com/financial/event/recorder/reader/Reader.kt

```diff
@@ -0,0 +1,13 @@
+package ar.com.financial.event.recorder.reader
+
+interface Reader<T> {
+
+    fun start()
+
+    fun stop()
+
+    fun hasNext(): Boolean
```

Comment:
Entiendo el concepto de un reader. Lo que a veces me pregunto es la necesidad de hasNext().
Me suena redundante a veces `if hasNext() then read()` cuando podría decir `read() and if not null do()` 
En Kotlin esto ultimo seria: `eventReader.read()?.let { eventWriter.write(it) }`